sets/maps.go

```diff
@@ -44,8 +44,14 @@ func IntersectMapKeys[K comparable, V any](maps ...map[K]V) map[K]V {
 	}
 	smallestMap := maps[smallestIdx]
 	final := make(map[K]V, len(smallestMap))
-	for k, v := range smallestMap {
-		final[k] = v
+	for k := range smallestMap {
+		for _, m := range maps {
```

Comment:
There is no need to iterate over all maps, if a value is in the intersection, it will appear in the 1st map